src/core/server/queue/tasks/mailer/index.ts

```diff
@@ -175,20 +58,28 @@ export class MailerQueue {
       return;
     }
 
-    // Generate the HTML for the email template.
-    const html = this.content.generateHTML({
-      ...template,
-      context: {
-        ...template.context,
-        tenant,
-      },
-    });
+    const startTemplateGenerationTime = now();
+
+    let html: string;
+    try {
+      // Generate the HTML for the email template.
+      html = await this.content.generateHTML(tenant, template);
+    } catch (err) {
+      log.error({ err }, "could not generate the html");
+      // TODO: (wyattjoh) maybe throw an error here?
```

Comment:
The intended behaviour or reasoning around throwing the error is that we would want to have the user retry the operation. This isn't always the case though, and logging it may trigger the admin's to repair the issue, and then re-trigger the required behaviour after the fact. If you still agree re-throwing here would be worth it, it can be done.

Reply:
Ok no that's fine :-)